pkg/workloads/console/controller.go

```diff
@@ -141,50 +147,37 @@ func (r *ConsoleReconciler) Reconcile(request reconcile.Request) (res reconcile.
 				"user", csl.Spec.User,
 			)
 
-			// Now that we've spawned a console job we can update the status, as
-			// the expiry and phase will have changed.
-			csl, err = r.updateStatus(csl, job)
-			if err != nil {
-				return res, err
-			}
-
-			// Requeue a reconciliation for when we expect the console expiry to fire
-			res = requeueForExpiration(logger, csl.Status)
 		} else {
 			logger.Log(
 				"event", EventExpired,
 				"resource", Job,
 				"name", name.Name,
 				"msg", "Not creating new job for expired console",
 			)
-			return res, nil
 		}
-
-		return res, err
+	} else {
+		// The console already exists
+		logger.Log(
+			"event", EventFound,
+			"resource", Job,
+			"name", name.Name,
+			"user", csl.Spec.User,
+		)
 	}
 
-	// The console already exists
-	logger.Log(
-		"event", EventFound,
-		"resource", Job,
-		"name", name.Name,
-		"user", csl.Spec.User,
-	)
-
 	// Find or create the role and role bindings
 	if err := r.updateRoleBindings(csl, consoleTemplate, name); err != nil {
 		return res, err
 	}
 
 	// Update the status fields in case they're out of sync, or the console spec
 	// has been updated
-	csl, err = r.updateStatus(csl, job)
-	if err != nil {
+	if csl, err = r.updateStatus(csl, job); err != nil {
```

Comment:
Keep in mind that the definition of `csl` here doesn't escape this block, so any further references in this function will be to the original value.

Reply:
```go
# main.go
package main

import (
	"fmt"

	"github.com/davecgh/go-spew/spew"
)

type obj struct {
	msg string
}

func newObj(msg string) *obj { return &obj{msg} }

func main() {
	o := &obj{"hello"}
	if o = newObj("world"); o != nil {
		fmt.Println("executing if")
	}

	spew.Dump(o)
}
```
```
λ (lab) ~> go run main.go
executing if
(*main.obj)(0xc000078290)({
 msg: (string) (len=5) "world"
})
```

Reply:
Ah, that might be right. Really fucking annoying how subtle this thing is.